enarx-pr-assigned

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/python3
+# SPDX-License-Identifier: Apache-2.0
+
+from github import Github
+import os
+
+RESPONSIBILITIES = {
+    'CHANGES_REQUESTED': lambda author, reviewer, requests: reviewer if reviewer in requests else author,
+    'DISMISSED': lambda author, reviewer, requests: reviewer,
+    'APPROVED': lambda author, reviewer, requests: None,
+    'COMMENTED': lambda author, reviewer, requests: reviewer,
+    'PENDING': lambda author, reviewer, requests: reviewer,
+}
+
+github = Github(os.environ['GITHUB_TOKEN'])
+
+# Return a dictionary with the latest review for every reviewer on a given PR.
+def get_latest_reviews(pr):
+    return {r.user.id: r for r in pr.get_reviews()}
+
+def get_requested_reviewers(pr):
+    # Collect sets of the latest reviews.
+    latest_reviews = get_latest_reviews(pr)
+    changes_requested_reviews = {u.user.id: u for (k, u) in latest_reviews.items() if u.state == "CHANGES_REQUESTED"}
+    
+    # Yield reviewers whose Changes Requested reviews have been addressed.
+    for e in pr.get_issue_events():
+        if(e.requested_reviewer is not None and
+        e.requested_reviewer.id in changes_requested_reviews and
+        e.created_at > changes_requested_reviews[e.requested_reviewer.id].submitted_at):
```

Comment:
Break up this condition by exclusion. For example: `if e.requested_reviewer is None: continue`. After all the conditionals, you can yield the event that made it through the filters.